span.go

```diff
@@ -65,10 +65,25 @@ func (tx *Transaction) StartSpan(name, spanType string, parent *Span) *Span {
 // span type, subtype, and action; a single dot separates span type and
 // subtype, and the action will not be set.
 func (tx *Transaction) StartSpanOptions(name, spanType string, opts SpanOptions) *Span {
-	if tx == nil || opts.parent.IsExitSpan() {
+	if tx == nil {
 		return newDroppedSpan()
 	}
 
+	if opts.parent.IsExitSpan() {
+		arr := strings.SplitN(spanType, ".", 3)
+
+		spanType := arr[0]
+		subType := ""
+
+		if len(arr) > 1 {
+			subType = arr[1]
+		}
```

Comment:
There's already code for splitting type/subtype/action. Should we extract that into a function that can be reused here?

https://github.com/elastic/apm-agent-go/blob/31ed25dde1519583f6a6160f47976be428bace87/span.go#L240-L246

Reply:
This no longer applies